modules/qna/src/views/full/Components/TextareaList.tsx

```diff
@@ -27,59 +26,50 @@ interface Props {
 
 const TextAreaList: FC<Props> = props => {
   const [showPicker, setShowPicker] = useState(false)
-  const {
-    items,
-    duplicateMsg,
-    updateItems,
-    keyPrefix,
-    canAddContent,
-    addItemLabel,
-    label,
-    refItems,
-    placeholder
-  } = props
+  const [localItems, setLocalItems] = useState(props.items)
+  // Generating unique keys so we don't need to rerender all the list as soon as we add or delete one element
+  const [keys, setKeys] = useState(localItems.map(x => _uniqueId(keyPrefix)))
+  const { duplicateMsg, updateItems, keyPrefix, canAddContent, addItemLabel, label, refItems, placeholder } = props
   const focusedElement = useRef(props.initialFocus || '')
 
   useEffect(() => {
-    keys.current = items.map(x => _uniqueId(keyPrefix))
+    setKeys(localItems.map(x => _uniqueId(keyPrefix)))
   }, [refItems])
 
-  // Generating unique keys so we don't need to rerender all the list as soon as we add or delete one element
-  const keys = useRef(items.map(x => _uniqueId(keyPrefix)))
-
-  const updateItem = (index: number, value: string): void => {
-    items[index] = value
-    updateItems(items)
+  const updateLocalItem = (index: number, value: string): void => {
+    localItems[index] = value
+    setLocalItems([...localItems])
   }
 
-  const debounceUpdateItem = useCallback(debounce(updateItem, 300), [])
-
   const addItem = (value = ''): void => {
-    items.push(value)
-    keys.current.push(_uniqueId(keyPrefix))
-    focusedElement.current = `${keyPrefix}${items.length - 1}`
-    updateItems(items)
+    localItems.push(value)
+    setKeys([...keys, _uniqueId(keyPrefix)])
+    focusedElement.current = `${keyPrefix}${localItems.length - 1}`
+    updateItems(localItems)
   }
 
   const onKeyDown = (e: KeyboardEvent, index: number): void => {
     if (e.key === 'Enter' && (e.ctrlKey || e.metaKey)) {
       e.preventDefault()
       addItem()
     }
-
-    if (e.key === 'Backspace' && items.length > 1 && !items[index].length) {
+    if (e.key === 'Backspace' && localItems.length > 1 && !localItems[index].length) {
```

Comment:
ok I understand what's that for but it took me some time, can you extract that condition in a variable like shouldDelete or delEmptyItem or whatever